payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayPaymentMethodLauncherViewModel.kt

```diff
@@ -0,0 +1,169 @@
+package com.stripe.android.googlepaylauncher
+
+import android.app.Application
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.distinctUntilChanged
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.wallet.PaymentData
+import com.google.android.gms.wallet.PaymentDataRequest
+import com.google.android.gms.wallet.PaymentsClient
+import com.stripe.android.GooglePayConfig
+import com.stripe.android.GooglePayJsonFactory
+import com.stripe.android.Logger
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.model.PaymentMethodCreateParams
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.networking.StripeApiRepository
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.paymentsheet.DefaultGooglePayRepository
+import com.stripe.android.paymentsheet.GooglePayRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import org.json.JSONObject
+import kotlin.coroutines.CoroutineContext
+
+internal class GooglePayPaymentMethodLauncherViewModel(
+    private val paymentsClient: PaymentsClient,
+    private val requestOptions: ApiRequest.Options,
+    private val args: GooglePayPaymentMethodLauncherContract.Args,
+    private val stripeRepository: StripeRepository,
+    private val googlePayJsonFactory: GooglePayJsonFactory,
+    private val googlePayRepository: GooglePayRepository
+) : ViewModel() {
+    var hasLaunched: Boolean = false
+
+    private val _googleResult = MutableLiveData<GooglePayPaymentMethodLauncher.Result>()
+    internal val googlePayResult = _googleResult.distinctUntilChanged()
+
+    fun updateResult(result: GooglePayPaymentMethodLauncher.Result) {
+        _googleResult.value = result
+    }
+
+    @VisibleForTesting
```

Comment:
I wonder if it would be worth putting in @VisibleForTesting(otherwise=VisibleForTesting.PRIVATE) or whatever?

Reply:
I think this is the default value.